utils/actionstack/actionstack.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2017 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package actionstack
+
+// Action represents stackable action.
+type Action func()
+
+// ActionStack is a stack of actions which are executed in reverse order
+// they were added
+type ActionStack struct {
+	stack []Action
+}
+
+// NewActionStack creates empty ActionStack
+func NewActionStack() *ActionStack {
+	return new(ActionStack)
+}
+
+// Run executes pushed actions in reverse order (FILO)
+func (a *ActionStack) Run() {
```

Comment:
Should we protect `Run` from being executed more than once?

Reply:
I think we should panic if run more than once or action was added after Run. Going to add it.

Reply:
Applied

Reply:
I'd rather we'd returned an error instead of a panic.

Reply:
It's essentially same situation when someone tries to take Lock twice - should never happen and if happens something is terribly wrong. But those who want call Run carelessly may wrap it with utils.CallOnce decorator.